freqtrade/exchange/exchange.py

```diff
@@ -677,33 +677,59 @@ def price_get_one_pip(self, pair: str, price: float) -> float:
         else:
             return 1 / pow(10, precision)
 
-    def get_min_pair_stake_amount(self, pair: str, price: float, stoploss: float,
-                                  leverage: Optional[float] = 1.0) -> Optional[float]:
+    def get_min_pair_stake_amount(
+        self,
+        pair: str,
+        price: float,
+        stoploss: float,
+        leverage: Optional[float] = 1.0
+    ) -> Optional[float]:
+        return self._get_stake_amount_limit(pair, price, stoploss, 'min', leverage)
+
+    def get_max_pair_stake_amount(
+        self,
+        pair: str,
+        price: float,
+        stoploss: float
+    ) -> Optional[float]:
+        return self._get_stake_amount_limit(pair, price, stoploss, 'max')
+
+    def _get_stake_amount_limit(
+        self,
+        pair: str,
+        price: float,
+        stoploss: float,
+        limit: Literal['min', 'max'],
+        leverage: Optional[float] = 1.0
```

Comment:
I would maybe add a `*,` argument here - forcing the use of keyword arguments after that - that'll make sure it's not called wrongly / with mistaking arguments

that's a nitpick though - overall i like this concept :+1:

Reply:
I don't really understand this, doesn't a star argument mean you can pass a variable number of arguments in?

Reply:
there's a difference - `*args, **kwargs` simply collects "all" arguments provided, and provides `args` as list, and `kwargs` as dict (with provided keywordarguments as content).

if you define a function as this:
`def whatever(*, pair, x):` - then all arguments after the * are required as kwargs
so a call `whatever(pair, x)` will be a syntaxerror, while `whatever(pair=pair, x=x)` will  work (it's the only way to have this work, actually).